xmipp

```diff
@@ -1194,8 +1197,32 @@ def getVersion(onlyNumbers=False):
     return cleanInfo.strip(' ')
 
 
+def ensureCompilerVersion(compiler):
+    if 'g++' in compiler or 'gcc' in compiler:
+        ensureGCC_GPPVersion(compiler)
+    else:
+        print(red('Version detection for \'' + compiler + '\' is not implemented.'))
+
+
+def ensureGCC_GPPVersion(compiler):
+    if 'TRAVIS' in os.environ:
+        return  # skip detection on TRAVIS
+    log = []
+    if not checkProgram(compiler, True):
+        sys.exit(-7)
+    runJob(compiler + " --version", show_output=False, show_command=False, log=log)
+    # expected first line format:
+    # gcc (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4
+    full_version = log[0].split()[3].strip()  # get 4.8.4 text, notice extra space after OS name
```

Comment:
This expectation is not so general. In Centos is broken. I suggest:
```
runJob(compiler + " -dumpversion", show_output=False, show_command=False, log=log)
full_version = log[0]
```
since with the `-dumpversion` only the version number is returned, i.e. `4.8.5`

Reply:
nice!
will do